internal/restorer/progressprinter/progressprinter.go

```diff
@@ -0,0 +1,113 @@
+package progressprinter
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sync"
+	"time"
+)
+
+var progresswriter io.Writer = os.Stdout
```

Comment:
access to os.Stdout should be restricted to `cmd/restic` as far as possible. Otherwise we lose the central control about which parts of restic generate output or not.

Instead please use ui/termstatus.Terminal. That struct already supports updating multiple status lines, non-terminal outputs, too long lines, is far more efficient and avoids duplicating code.